core/account-id/src/lib.rs

```diff
@@ -9,23 +72,177 @@ mod serde;
 
 #[cfg(feature = "deepsize_feature")]
 use deepsize::DeepSizeOf;
-pub use error::{ParseAccountError, ParseErrorKind};
+pub use errors::{ParseAccountError, ParseErrorKind};
 
+/// Smallest valid length for a NEAR Account ID.
 pub const MIN_ACCOUNT_ID_LEN: usize = 2;
+/// Largest valid length for a NEAR Account ID.
 pub const MAX_ACCOUNT_ID_LEN: usize = 64;
 
-/// Account identifier. Provides access to user's state.
+/// NEAR Account Identifier.
+///
+/// This is a unique, validated, human-readable account identifier on the NEAR network.
+///
+/// [See the crate-level docs for information about validation.](index.html#account-id-rules)
+///
+/// ## Examples
+///
+/// ```
+/// use near_account_id::AccountId;
 ///
-/// This guarantees all properly constructed AccountId's are valid for the NEAR network.
+/// let alice: AccountId = "alice.near".parse().unwrap();
+///
+/// // Basic reports for why validation failed
+/// assert!(
+///   matches!(
+///     "z".parse::<AccountId>(),
+///     Err(err) if err.kind().is_too_short()
+///   )
+/// );
+///
+/// assert!(
+///   matches!(
+///     // no caps
+///     "MelissaCarver.near".parse::<AccountId>(),
+///     Err(err) if err.kind().is_invalid()
+///   )
+/// );
+///
+/// assert!(
+///   matches!(
+///     // separators cannot immediately follow each other
+///     "bob__carol".parse::<AccountId>(),
+///     Err(err) if err.kind().is_invalid()
+///   )
+/// );
+///
+/// assert!(
+///   matches!(
+///     // each part must be alphanumeric only (ƒ is not f)
+///     "ƒelicia.near".parse::<AccountId>(),
+///     Err(err) if err.kind().is_invalid()
+///   )
+/// );
+/// ```
 #[cfg_attr(feature = "deepsize_feature", derive(DeepSizeOf))]
 #[derive(Eq, Ord, Hash, Clone, Debug, PartialEq, PartialOrd)]
 pub struct AccountId(Box<str>);
 
 impl AccountId {
+    /// Returns the length of the Account ID.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use near_account_id::AccountId;
+    ///
+    /// let carol: AccountId = "carol.near".parse().unwrap();
+    /// assert_eq!(10, carol.len());
+    /// ```
+    #[allow(clippy::len_without_is_empty)]
     pub fn len(&self) -> usize {
```

Comment:
I am curious why we have this specific method, but not all other methods available on `&str`?

Reply:
Implemented `Deref<str>`.